e2e/test/scripts/create_cluster.sh

```diff
@@ -3,39 +3,68 @@
 set -o errexit
 set -o pipefail
 set -o nounset
+set -x
 
-export LINODE_API_TOKEN="$1"
+export LINODE_TOKEN="$1"
 export CLUSTER_NAME="$2"
-export IMAGE="$3"
-export K8S_VERSION="$4"
+export IMG="$3"
+export KUBERNETES_VERSION="$4"
+export CAPL_VERSION="0.1.0"
+export WORKER_MACHINE_COUNT=2
+export LINODE_CONTROL_PLANE_MACHINE_TYPE=g6-standard-2
+export LINODE_MACHINE_TYPE=g6-standard-2
+export KUBECONFIG="$(realpath "$(dirname "$0")/../kind-management.conf")"
+export ROOT_DIR="$(git rev-parse --show-toplevel)"
 
 if [[ -z "$5" ]]
 then
-  export REGION="eu-west"
+  export LINODE_REGION="us-sea"
 else
-  export REGION="$5"
+  export LINODE_REGION="$5"
 fi
 
-cat > cluster.tf <<EOF
-variable "nodes" {
-  default = 2
-}
+METADATA_REGIONS="nl-ams in-maa us-ord id-cgk us-lax es-mad us-mia it-mil jp-osa fr-par br-gru us-sea se-sto us-iad"
+[[ "$METADATA_REGIONS" =~ .*"${LINODE_REGION}".* ]] || (echo "Given region doesn't support Metadata service" ; exit 1)
 
-module "k8s" {
-  source       = "git::https://github.com/linode/terraform-linode-k8s.git"
-  k8s_version  = "${K8S_VERSION}"
-  linode_token = "${LINODE_API_TOKEN}"
-  ccm_image    = "${IMAGE}"
-  region       = "${REGION}"
-  cluster_name = "${CLUSTER_NAME}"
-  nodes        = var.nodes
-}
-EOF
+(cd ${ROOT_DIR}/deploy ; set +x ; ./generate-manifest.sh ${LINODE_TOKEN} ${LINODE_REGION})
 
-terraform workspace new ${CLUSTER_NAME}
+kubectl create ns ${CLUSTER_NAME}
+(cd $(realpath "$(dirname "$0")"); clusterctl generate cluster ${CLUSTER_NAME} \
+  --target-namespace ${CLUSTER_NAME} \
+  --flavor clusterclass-kubeadm \
+  --config clusterctl.yaml \
+  | kubectl apply --wait -f -)
 
-terraform init
+c=8
+until kubectl get secret -n ${CLUSTER_NAME} ${CLUSTER_NAME}-kubeconfig; do
+  sleep $(((c--)))
+done
 
-terraform apply -auto-approve
+kubectl get secret -n ${CLUSTER_NAME} ${CLUSTER_NAME}-kubeconfig -o jsonpath="{.data.value}" \
+  | base64 --decode \
+  > "$(pwd)/${CLUSTER_NAME}.conf"
 
 export KUBECONFIG="$(pwd)/${CLUSTER_NAME}.conf"
+
+c=16
+until kubectl version; do
+  sleep $(((c--)))
+done
+
+kubectl get no -l node-role.kubernetes.io/control-plane --no-headers -o custom-columns=":metadata.name" \
```

Comment:
The `node-role.kubernetes.io/master` label is hard coded, but CAPL clusters don't have this label. Interestingly, because the node selector of daemonset is `node-role.kubernetes.io/control-plane`.

For me this looks typo in test framework at https://github.com/linode/linode-cloud-controller-manager/blob/4c7159ed22f29430ae34457fce3b2f70176fd250/e2e/test/framework/node.go#L10

Reply:
That label is [deprecated](https://kubernetes.io/docs/reference/labels-annotations-taints/#node-role-kubernetes-io-master-taint)